modules/polyfills/src/fetch-node/read-file.node.js

```diff
@@ -30,9 +30,32 @@ export async function readFile(url, options = {}) {
 
   if (isRequestURL(url)) {
     return new Promise((resolve, reject) => {
-      options = {...new URL(url), ...options};
+      const urlObject = new URL(url);
+      let requestOptions = Object.assign(
+        {},
+        {
+          hostname: urlObject.hostname,
+          path: urlObject.pathname,
+          method: 'GET'
+        },
+        // Add options and user provided 'options.fetch' overrides if available
+        options,
+        options.fetch,
+        // Inject accepted encodings:
+        {
+          headers: {'Accept-Encoding': 'gzip,br,deflate'}
+        }
+      );
+      // And allow any overrides if needed by user
+      if (options.headers) {
```

Comment:
If the user supplies a header, the accept-encoding header no longer get injected. You could probably move the header calculation up before the options to handle that.

```
const headers = options.headers || {};
headers['Accept-Encoding'] =  headers['Accept-Encoding'] || 'gzip,br,deflate';
```

This is not perfect either, since headers should be independent of capitalization, but good enough for now.